From 6372d4a443de7aedbdec537d5437e02cb57aff55 Mon Sep 17 00:00:00 2001
From: Zongdong Jiao <zongdong.jiao@amlogic.com>
Date: Thu, 12 Feb 2015 14:41:44 +0800
Subject: [PATCH 5412/5965] PD#102054: hdmitx2: disable Dithering

The incoming video stream is 10bits, and using VPU_HDMI_FMT_CTRL[4]
to dither 10bits to 8bits.
Disable this function to avoid water-flickers.

Change-Id: I4634f4bb5f5b013f874906072b7bcff6e7dfa4cb
---
 arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c b/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
index e343b3596c4e..2134eb6b83a7 100755
--- a/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
+++ b/arch/arm/mach-mesong9tv/hdmi_tx_hw_20/hdmi_tx_hw.c
@@ -1856,7 +1856,7 @@ static int hdmitx_set_dispmode(hdmitx_dev_t* hdev, Hdmi_tx_video_para_t *param)
     }
     aml_write_reg32(P_VPU_HDMI_FMT_CTRL,(((TX_INPUT_COLOR_FORMAT==HDMI_COLOR_FORMAT_420)?2:0)  << 0) | // [ 1: 0] hdmi_vid_fmt. 0=444; 1=convert to 422; 2=convert to 420.
                          (2                                                     << 2) | // [ 3: 2] chroma_dnsmp. 0=use pixel 0; 1=use pixel 1; 2=use average.
-                         (((TX_COLOR_DEPTH==HDMI_COLOR_DEPTH_24B)? 1:0)         << 4) | // [    4] dith_en. 1=enable dithering before HDMI TX input.
+                         (0                                                     << 4) | // [    4] dith_en. 1=enable dithering before HDMI TX input.
                          (0                                                     << 5) | // [    5] hdmi_dith_md: random noise selector.
                          (0                                                     << 6)); // [ 9: 6] hdmi_dith10_cntl.
     if(hdev->mode420 == 1) {
@@ -2590,7 +2590,6 @@ static void hdmitx_4k2k5g420_debug(void)
     aml_write_reg32(P_HHI_VID_CLK_DIV, 0x100);
     aml_set_reg32_bits(P_HHI_HDMI_CLK_CNTL, 1, 16, 4);
     aml_write_reg32(P_VPU_HDMI_SETTING, 0x10e);
-    aml_write_reg32(P_VPU_HDMI_FMT_CTRL, 0x1a);
     hdmitx_wr_reg(HDMITX_DWC_FC_SCRAMBLER_CTRL, 0);
     hdmitx_wr_reg(HDMITX_TOP_TMDS_CLK_PTTN_01, 0x001f001f);
     printk("%s[%d]\n", __func__, __LINE__);
@@ -3108,7 +3107,7 @@ static int hdmitx_cntl_misc(hdmitx_dev_t* hdmitx_device, unsigned cmd, unsigned
         return !!(aml_read_reg32(P_HHI_VID2_PLL_CNTL) & (1 << 30));     // bit30: enable
         break;
     case MISC_CONF_MODE420:
-        aml_write_reg32(P_VPU_HDMI_FMT_CTRL, 0x1a);
+        aml_write_reg32(P_VPU_HDMI_FMT_CTRL, 0xa);
         aml_write_reg32(P_VPU_HDMI_SETTING, 0x10e);
         break;
     case MISC_TMDS_CLK_DIV40:
-- 
2.19.0

